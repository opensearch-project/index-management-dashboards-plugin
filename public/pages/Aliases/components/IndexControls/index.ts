/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import IndexControls from "./IndexControls";

export * from "./IndexControls";
export default IndexControls;
