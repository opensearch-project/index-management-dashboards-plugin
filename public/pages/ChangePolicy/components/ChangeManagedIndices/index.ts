/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import ChangeManagedIndices from "./ChangeManagedIndices";

export default ChangeManagedIndices;
