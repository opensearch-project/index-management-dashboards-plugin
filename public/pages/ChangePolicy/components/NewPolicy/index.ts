/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import NewPolicy from "./NewPolicy";

export default NewPolicy;
