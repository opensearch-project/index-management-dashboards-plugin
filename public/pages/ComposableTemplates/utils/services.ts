/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { CommonService } from "../../../services";
import { TemplateItemRemote } from "../../../../models/interfaces";

export const getAllUsedComponents = async ({ commonService }: { commonService: CommonService }) => {
  const allTemplatesResponse = await commonService.apiCaller<{
    index_templates?: Array<{
      name: string;
      index_template: TemplateItemRemote;
    }>;
  }>({
    endpoint: "transport.request",
    data: {
      method: "GET",
      path: "/_index_template/*",
    },
  });

  if (allTemplatesResponse && allTemplatesResponse.ok) {
    return allTemplatesResponse.response.index_templates || [];
  }

  return [];
};
