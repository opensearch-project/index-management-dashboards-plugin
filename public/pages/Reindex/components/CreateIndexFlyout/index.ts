/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import CreateIndexFlyout from "./CreateIndexFlyout";

export default CreateIndexFlyout;
