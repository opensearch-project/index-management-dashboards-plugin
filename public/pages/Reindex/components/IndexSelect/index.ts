/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import IndexSelect from "./IndexSelect";

export default IndexSelect;
