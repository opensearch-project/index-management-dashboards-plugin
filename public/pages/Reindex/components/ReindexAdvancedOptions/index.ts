/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import ReindexAdvancedOptions from "./ReindexAdvancedOptions";

export default ReindexAdvancedOptions;
