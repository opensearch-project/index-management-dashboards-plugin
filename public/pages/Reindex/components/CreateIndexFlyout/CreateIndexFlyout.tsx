/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import CreateIndex, { IndexFormProps } from "../../../CreateIndex/containers/IndexForm";
import { CoreServicesContext } from "../../../../components/core_services";
import { EuiFlyout, EuiFlyoutBody, EuiFlyoutFooter, EuiFlyoutHeader, EuiTitle } from "@elastic/eui";
import FlyoutFooter from "../../../VisualCreatePolicy/components/FlyoutFooter";

interface CreateIndexFlyoutProps extends IndexFormProps {
  sourceIndices: string[];
  onCloseFlyout: () => void;
}

export default class CreateIndexFlyout extends React.Component<CreateIndexFlyoutProps> {
  static contextType = CoreServicesContext;

  createIndexRef: CreateIndex | null = null;

  render() {
    const { onCloseFlyout } = this.props;
    return (
      <EuiFlyout onClose={() => {}} hideCloseButton>
        <EuiFlyoutHeader hasBorder>
          <EuiTitle size="m">
            <h2 id="flyoutTitle"> Create Index </h2>
          </EuiTitle>
        </EuiFlyoutHeader>

        <EuiFlyoutBody>
          <CreateIndex ref={(ref) => (this.createIndexRef = ref)} hideButtons={true} {...this.props} />
        </EuiFlyoutBody>

        <EuiFlyoutFooter>
          <FlyoutFooter
            action=""
            text="Create"
            edit={false}
            onClickCancel={onCloseFlyout}
            onClickAction={() => this.createIndexRef?.onSubmit()}
          />
        </EuiFlyoutFooter>
      </EuiFlyout>
    );
  }
}
