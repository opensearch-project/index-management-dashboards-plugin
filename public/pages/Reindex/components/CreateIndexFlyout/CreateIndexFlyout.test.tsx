/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { render, waitFor } from "@testing-library/react";
import React from "react";
import CreateIndexFlyout from "./CreateIndexFlyout";
import { coreServicesMock, browserServicesMock, apiCallerMock } from "../../../../../test/mocks";
import { CoreServicesContext } from "../../../../components/core_services";
import { ServicesContext } from "../../../../services";

describe("<CreateIndexFlyout /> spec", () => {
  beforeEach(() => {
    apiCallerMock(browserServicesMock);
  });
  it("renders the component", async () => {
    const component = render(
      <CoreServicesContext.Provider value={coreServicesMock}>
        <ServicesContext.Provider value={browserServicesMock}>
          <CreateIndexFlyout sourceIndices={[]} onCloseFlyout={() => {}} />,
        </ServicesContext.Provider>
      </CoreServicesContext.Provider>
    );
    await waitFor(
      () => expect((document.querySelector("#accordionForCreateIndexSettings") as HTMLDivElement).style.height).toEqual("0px"),
      {
        timeout: 3000,
      }
    );
    expect(component).toMatchSnapshot();
  }, 20000);
});
