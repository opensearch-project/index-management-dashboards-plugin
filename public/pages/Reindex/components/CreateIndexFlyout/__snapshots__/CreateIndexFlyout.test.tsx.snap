// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateIndexFlyout /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="euiBody-hasOverlayMask euiBody--hasFlyout"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    >
      ,
    </div>
    <div
      class="euiOverlayMask euiOverlayMask--belowHeader"
    >
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
      <div
        data-focus-lock-disabled="false"
      >
        <div
          class="euiFlyout euiFlyout--medium euiFlyout--paddingLarge"
          role="dialog"
          tabindex="-1"
        >
          <div
            class="euiFlyoutHeader euiFlyoutHeader--hasBorder"
          >
            <h2
              class="euiTitle euiTitle--medium"
              id="flyoutTitle"
            >
               Create Index 
            </h2>
          </div>
          <div
            class="euiFlyoutBody"
          >
            <div
              class="euiFlyoutBody__overflow"
              tabindex="0"
            >
              <div
                class="euiFlyoutBody__overflowContent"
              >
                <div
                  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
                >
                  <div
                    class="euiText euiText--small"
                  >
                     
                    <h2>
                      Define index
                    </h2>
                     
                  </div>
                  <hr
                    class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
                  />
                  <div
                    class="euiForm"
                  >
                    <div
                      class="euiFormRow euiFormRow--compressed"
                      data-test-subj="form-name-index"
                      id="some_html_id-row"
                    >
                      <div
                        class="euiFormRow__labelWrapper"
                      >
                        <label
                          aria-invalid="false"
                          class="euiFormLabel euiFormRow__label"
                          for="some_html_id"
                        >
                          <div
                            class="euiText euiText--small"
                          >
                            <h3>
                              Index name
                            </h3>
                          </div>
                        </label>
                      </div>
                      <div
                        class="euiFormRow__fieldWrapper"
                      >
                        <span
                          class="euiToolTipAnchor euiToolTipAnchor--displayBlock"
                        >
                          <div
                            class="euiFormControlLayout euiFormControlLayout--compressed"
                          >
                            <div
                              class="euiFormControlLayout__childrenWrapper"
                            >
                              <input
                                class="euiFieldText euiFieldText--compressed"
                                placeholder="Specify a name for the new index."
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </span>
                        <div
                          class="euiFormHelpText euiFormRow__text"
                          id="some_html_id-help-0"
                        >
                          <div>
                            Must be in lowercase letters. Cannot begin with underscores or hyphens. Spaces, commas, and characters :, ", *, +, /, , |, ?, #, &gt; are not allowed.
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="euiFormRow euiFormRow--fullWidth euiFormRow--compressed"
                      data-test-subj="form-name-templateMessage"
                      id="some_html_id-row"
                    >
                      <div
                        class="euiFormRow__fieldWrapper"
                      />
                    </div>
                    <div
                      class="euiFormRow euiFormRow--compressed"
                      data-test-subj="form-name-aliases"
                      id="some_html_id-row"
                    >
                      <div
                        class="euiFormRow__labelWrapper"
                      >
                        <label
                          aria-invalid="false"
                          class="euiFormLabel euiFormRow__label"
                          for="some_html_id"
                        >
                          <div
                            class="euiText euiText--small"
                          >
                            <h3>
                              Index alias 
                              <i>
                                 â€“ optional
                              </i>
                            </h3>
                          </div>
                        </label>
                      </div>
                      <div
                        class="euiFormRow__fieldWrapper"
                      >
                        <div
                          class="euiFormHelpText euiFormRow__text"
                          style="padding-top: 0px; padding-bottom: 4px;"
                        >
                          Allow this index to be referenced by existing aliases or specify a new alias.
                        </div>
                        <span
                          class="euiToolTipAnchor euiToolTipAnchor--displayBlock"
                        >
                          <div
                            aria-expanded="false"
                            aria-haspopup="listbox"
                            class="euiComboBox euiComboBox--compressed"
                            role="combobox"
                          >
                            <div
                              class="euiFormControlLayout euiFormControlLayout--compressed"
                            >
                              <div
                                class="euiFormControlLayout__childrenWrapper"
                              >
                                <div
                                  class="euiComboBox__inputWrap euiComboBox__inputWrap--compressed euiComboBox__inputWrap-isClearable"
                                  data-test-subj="comboBoxInput"
                                  tabindex="-1"
                                >
                                  <p
                                    class="euiComboBoxPlaceholder"
                                  >
                                    Select aliases or specify new aliases.
                                  </p>
                                  <div
                                    class="euiComboBox__input"
                                    style="font-size: 14px; display: inline-block;"
                                  >
                                    <input
                                      aria-controls=""
                                      data-test-subj="comboBoxSearchInput"
                                      role="textbox"
                                      style="box-sizing: content-box; width: 2px;"
                                      value=""
                                    />
                                    <div
                                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                                    />
                                  </div>
                                </div>
                                <div
                                  class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                                >
                                  <button
                                    aria-label="Open list of options"
                                    class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                                    data-test-subj="comboBoxToggleListButton"
                                    type="button"
                                  >
                                    EuiIconMock
                                  </button>
                                </div>
                              </div>
                            </div>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="euiSpacer euiSpacer--l"
                />
                <div
                  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
                >
                  <div
                    class="euiText euiText--small"
                  >
                     
                    <h2>
                      Index settings
                    </h2>
                     
                  </div>
                  <hr
                    class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
                  />
                  <div
                    class="euiForm"
                  >
                    <div
                      class="euiFormRow euiFormRow--compressed"
                      data-test-subj="form-name-index.number_of_shards"
                      id="some_html_id-row"
                    >
                      <div
                        class="euiFormRow__labelWrapper"
                      >
                        <label
                          aria-invalid="false"
                          class="euiFormLabel euiFormRow__label"
                          for="some_html_id"
                        >
                          <div
                            class="euiText euiText--small"
                          >
                            <h3>
                              Number of primary shards
                            </h3>
                          </div>
                        </label>
                      </div>
                      <div
                        class="euiFormRow__fieldWrapper"
                      >
                        <div
                          class="euiFormHelpText euiFormRow__text"
                          style="padding-top: 0px; padding-bottom: 4px;"
                        >
                          <div>
                            Specify the number of primary shards for the index. Default is 1. 
                          </div>
                          <div>
                            The number of primary shards cannot be changed after the index is created.
                          </div>
                        </div>
                        <span
                          class="euiToolTipAnchor euiToolTipAnchor--displayBlock"
                        >
                          <div
                            class="euiFormControlLayout euiFormControlLayout--compressed"
                          >
                            <div
                              class="euiFormControlLayout__childrenWrapper"
                            >
                              <input
                                class="euiFieldNumber euiFieldNumber--compressed"
                                placeholder="Specify primary shard count."
                                type="number"
                                value="1"
                              />
                            </div>
                          </div>
                        </span>
                      </div>
                    </div>
                    <div
                      class="euiFormRow euiFormRow--compressed"
                      data-test-subj="form-name-index.number_of_replicas"
                      id="some_html_id-row"
                    >
                      <div
                        class="euiFormRow__labelWrapper"
                      >
                        <label
                          aria-invalid="false"
                          class="euiFormLabel euiFormRow__label"
                          for="some_html_id"
                        >
                          <div
                            class="euiText euiText--small"
                          >
                            <h3>
                              Number of replicas
                            </h3>
                          </div>
                        </label>
                      </div>
                      <div
                        class="euiFormRow__fieldWrapper"
                      >
                        <div
                          class="euiFormHelpText euiFormRow__text"
                          style="padding-top: 0px; padding-bottom: 4px;"
                        >
                          Specify the number of replicas each primary shard should have. Default is 1.
                        </div>
                        <span
                          class="euiToolTipAnchor euiToolTipAnchor--displayBlock"
                        >
                          <div
                            class="euiFormControlLayout euiFormControlLayout--compressed"
                          >
                            <div
                              class="euiFormControlLayout__childrenWrapper"
                            >
                              <input
                                class="euiFieldNumber euiFieldNumber--compressed"
                                placeholder="Specify number of replicas."
                                type="number"
                                value="1"
                              />
                            </div>
                          </div>
                        </span>
                      </div>
                    </div>
                    <div
                      class="euiFormRow euiFormRow--compressed"
                      data-test-subj="form-name-index.refresh_interval"
                      id="some_html_id-row"
                    >
                      <div
                        class="euiFormRow__labelWrapper"
                      >
                        <label
                          aria-invalid="false"
                          class="euiFormLabel euiFormRow__label"
                          for="some_html_id"
                        >
                          <div
                            class="euiText euiText--small"
                          >
                            <h3>
                              Refresh interval
                            </h3>
                          </div>
                        </label>
                      </div>
                      <div
                        class="euiFormRow__fieldWrapper"
                      >
                        <div
                          class="euiFormHelpText euiFormRow__text"
                          style="padding-top: 0px; padding-bottom: 4px;"
                        >
                          Specify how often the index should refresh, which publishes the most recent changes and make them available for search. Default is 1 second.
                        </div>
                        <span
                          class="euiToolTipAnchor euiToolTipAnchor--displayBlock"
                        >
                          <div
                            class="euiFormControlLayout euiFormControlLayout--compressed"
                          >
                            <div
                              class="euiFormControlLayout__childrenWrapper"
                            >
                              <input
                                class="euiFieldText euiFieldText--compressed"
                                placeholder="Can be set to -1 to disable refreshing."
                                type="text"
                                value="1s"
                              />
                            </div>
                          </div>
                        </span>
                      </div>
                    </div>
                    <div
                      class="euiSpacer euiSpacer--m"
                    />
                    <div
                      class="euiAccordion accordion-in-advanced-settings"
                    >
                      <div
                        class="euiAccordion__triggerWrapper"
                      >
                        <button
                          aria-controls="accordionForCreateIndexSettings"
                          aria-expanded="false"
                          class="euiAccordion__button"
                          id="some_html_id"
                          type="button"
                        >
                          <span
                            class="euiAccordion__iconWrapper"
                          >
                            EuiIconMock
                          </span>
                          <span
                            class="euiIEFlexWrapFix"
                          >
                            <h3>
                              Advanced settings
                            </h3>
                          </span>
                        </button>
                      </div>
                      <div
                        aria-labelledby="some_html_id"
                        class="euiAccordion__childWrapper"
                        id="accordionForCreateIndexSettings"
                        role="region"
                        style="height: 0px"
                        tabindex="-1"
                      >
                        <div>
                          <div
                            class=""
                          >
                            <div
                              class="euiSpacer euiSpacer--s"
                            />
                            <div
                              class="euiFormRow euiFormRow--fullWidth euiFormRow--compressed"
                              data-test-subj="formNameAdvancedSettings"
                              id="some_html_id-row"
                            >
                              <div
                                class="euiFormRow__labelWrapper"
                              >
                                <label
                                  class="euiFormLabel euiFormRow__label"
                                  for="some_html_id"
                                >
                                  Specify advanced index settings
                                </label>
                              </div>
                              <div
                                class="euiFormRow__fieldWrapper"
                              >
                                <div
                                  class="euiFormHelpText euiFormRow__text"
                                  style="padding-top: 0px; padding-bottom: 4px;"
                                >
                                  <p>
                                    Specify a comma-delimited list of settings.
                                     
                                    <a
                                      class="euiLink euiLink--primary"
                                      href="https://opensearch.org/docs/latest/api-reference/index-apis/create-index#index-settings"
                                      rel="noopener noreferrer"
                                      target="_blank"
                                    >
                                      View index settings.
                                      EuiIconMock
                                      <span
                                        class="euiScreenReaderOnly"
                                      >
                                        (opens in a new tab or window)
                                      </span>
                                    </a>
                                  </p>
                                  <p>
                                    All the settings will be handled in flat structure.
                                     
                                    <a
                                      class="euiLink euiLink--primary"
                                      href="https://opensearch.org/docs/undefined/api-reference/index-apis/get-index/#url-parameters"
                                      rel="noopener noreferrer"
                                      target="_blank"
                                    >
                                      Learn more
                                      EuiIconMock
                                      <span
                                        class="euiScreenReaderOnly"
                                      >
                                        (opens in a new tab or window)
                                      </span>
                                    </a>
                                    .
                                  </p>
                                </div>
                                <textarea
                                  data-test-subj="jsonEditor-valueDisplay"
                                  readonly=""
                                  style="display: none;"
                                >
                                  {
  "index.number_of_shards": 1,
  "index.number_of_replicas": 1,
  "index.refresh_interval": "1s"
}
                                </textarea>
                                <div
                                  class="euiCodeEditorWrapper"
                                  data-test-subj="codeEditorContainer"
                                  style="width: 100%;"
                                >
                                  <button
                                    class="euiCodeEditorKeyboardHint"
                                    data-test-subj="codeEditorHint"
                                    id="some_html_id"
                                  >
                                    <p
                                      class="euiText"
                                    >
                                      Press Enter to start editing.
                                    </p>
                                    <p
                                      class="euiText"
                                    >
                                      When you're done, press Escape to stop editing.
                                    </p>
                                  </button>
                                  <div
                                    class=" ace_editor ace-tm"
                                    id="some_html_id"
                                    style="width: 100%; height: 500px;"
                                  >
                                    <textarea
                                      autocapitalize="off"
                                      autocorrect="off"
                                      class="ace_text-input"
                                      spellcheck="false"
                                      style="opacity: 0;"
                                      tabindex="-1"
                                      wrap="off"
                                    />
                                    <div
                                      aria-hidden="true"
                                      class="ace_gutter"
                                    >
                                      <div
                                        class="ace_layer ace_gutter-layer ace_folding-enabled"
                                      />
                                      <div
                                        class="ace_gutter-active-line"
                                      />
                                    </div>
                                    <div
                                      class="ace_scroller"
                                    >
                                      <div
                                        class="ace_content"
                                      >
                                        <div
                                          class="ace_layer ace_print-margin-layer"
                                        >
                                          <div
                                            class="ace_print-margin"
                                            style="left: 4px; visibility: visible;"
                                          />
                                        </div>
                                        <div
                                          class="ace_layer ace_marker-layer"
                                        />
                                        <div
                                          class="ace_layer ace_text-layer"
                                          style="padding: 0px 4px;"
                                        />
                                        <div
                                          class="ace_layer ace_marker-layer"
                                        />
                                        <div
                                          class="ace_layer ace_cursor-layer ace_hidden-cursors"
                                        >
                                          <div
                                            class="ace_cursor"
                                          />
                                        </div>
                                      </div>
                                    </div>
                                    <div
                                      class="ace_scrollbar ace_scrollbar-v"
                                      style="display: none; width: 20px;"
                                    >
                                      <div
                                        class="ace_scrollbar-inner"
                                        style="width: 20px;"
                                      />
                                    </div>
                                    <div
                                      class="ace_scrollbar ace_scrollbar-h"
                                      style="display: none; height: 20px;"
                                    >
                                      <div
                                        class="ace_scrollbar-inner"
                                        style="height: 20px;"
                                      />
                                    </div>
                                    <div
                                      style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;"
                                    >
                                      <div
                                        style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
                                      />
                                      <div
                                        style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
                                      >
                                        XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="euiSpacer euiSpacer--l"
                />
                <div
                  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
                >
                  <div
                    class="euiText euiText--small"
                  >
                    <h2>
                      Index mapping
                      <i>
                         â€“ optional
                      </i>
                    </h2>
                  </div>
                  <div
                    class="euiFormRow euiFormRow--fullWidth euiFormRow--compressed"
                    id="some_html_id-row"
                  >
                    <div
                      class="euiFormRow__fieldWrapper"
                    >
                      <div
                        class="euiFormHelpText euiFormRow__text"
                        id="some_html_id-help-0"
                      >
                        <div>
                          Define how documents and their fields are stored and indexed.
                           
                          <a
                            class="euiLink euiLink--primary"
                            href="https://opensearch.org/docs/undefined/opensearch/mappings/"
                            rel="noopener noreferrer"
                            target="_blank"
                          >
                            Learn more
                            EuiIconMock
                            <span
                              class="euiScreenReaderOnly"
                            >
                              (opens in a new tab or window)
                            </span>
                          </a>
                        </div>
                        <div>
                          Mappings and field types cannot be changed after the index is created.
                        </div>
                      </div>
                    </div>
                  </div>
                  <hr
                    class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
                  />
                  <div
                    class="euiFormRow euiFormRow--fullWidth euiFormRow--compressed"
                    id="some_html_id-row"
                  >
                    <div
                      class="euiFormRow__fieldWrapper"
                    >
                      <fieldset
                        class="euiButtonGroup euiButtonGroup--small euiButtonGroup--text"
                      >
                        <legend
                          class="euiScreenReaderOnly"
                        >
                          Editor Type
                        </legend>
                        <div
                          class="euiButtonGroup__buttons"
                        >
                          <label
                            class="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small euiButtonGroupButton--fill euiButtonGroupButton-isSelected"
                            data-test-subj="editorTypeVisualEditor"
                            for="some_html_id"
                          >
                            <span
                              class="euiButtonContent euiButton__content"
                            >
                              <span
                                class="euiButton__text euiButtonGroupButton__textShift"
                                data-text="Visual Editor"
                                title="Visual Editor"
                              >
                                <input
                                  checked=""
                                  class="euiScreenReaderOnly"
                                  data-test-subj="VISUAL"
                                  id="some_html_id"
                                  name="some_html_id"
                                  type="radio"
                                  value=""
                                />
                                Visual Editor
                              </span>
                            </span>
                          </label>
                          <label
                            class="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small"
                            data-test-subj="editorTypeJsonEditor"
                            for="some_html_id"
                          >
                            <span
                              class="euiButtonContent euiButton__content"
                            >
                              <span
                                class="euiButton__text euiButtonGroupButton__textShift"
                                data-text="JSON Editor"
                                title="JSON Editor"
                              >
                                <input
                                  class="euiScreenReaderOnly"
                                  data-test-subj="JSON"
                                  id="some_html_id"
                                  name="some_html_id"
                                  type="radio"
                                  value=""
                                />
                                JSON Editor
                              </span>
                            </span>
                          </label>
                        </div>
                      </fieldset>
                      <div
                        class="euiSpacer euiSpacer--l"
                      />
                      <div
                        class="euiText euiText--small"
                      >
                        <p>
                          You have no field mappings.
                        </p>
                      </div>
                      <div
                        class="euiSpacer euiSpacer--l"
                      />
                      <button
                        class="euiButton euiButton--primary euiButton--small"
                        data-test-subj="createIndexAddFieldButton"
                        style="margin-right: 8px;"
                        type="button"
                      >
                        <span
                          class="euiButtonContent euiButton__content"
                        >
                          <span
                            class="euiButton__text"
                          >
                            Add new field
                          </span>
                        </span>
                      </button>
                      <button
                        class="euiButton euiButton--primary euiButton--small"
                        data-test-subj="createIndexAddObjectFieldButton"
                        type="button"
                      >
                        <span
                          class="euiButtonContent euiButton__content"
                        >
                          <span
                            class="euiButton__text"
                          >
                            Add new object
                          </span>
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiFlyoutFooter"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
            >
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <button
                  class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--small euiButtonEmpty--flushLeft"
                  data-test-subj="flyout-footer-cancel-button"
                  type="button"
                >
                  <span
                    class="euiButtonContent euiButtonEmpty__content"
                  >
                    <span
                      class="euiButtonEmpty__text"
                    >
                      Cancel
                    </span>
                  </span>
                </button>
              </div>
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <button
                  class="euiButton euiButton--primary euiButton--small euiButton--fill"
                  data-test-subj="flyout-footer-action-button"
                  type="button"
                >
                  <span
                    class="euiButtonContent euiButton__content"
                  >
                    <span
                      class="euiButton__text"
                    >
                      Create
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
